src/v/kafka/client/cluster.cc

```diff
@@ -370,13 +371,39 @@ void cluster::update_configuration(connection_configuration config) {
         set_sasl_configuration(std::move(config.sasl_cfg));
     }
 
+    if (
+      config.initial_brokers != _config.initial_brokers
+      || config.broker_tls != _config.broker_tls) {
+        ssx::background
+          = update_broker_config(
+              std::move(config.initial_brokers), std::move(config.broker_tls))
+              .handle_exception([this](std::exception_ptr ex) {
+                  vlog(_logger.warn, "Failed to update broker config: {}", ex);
+              });
+    }
+
```

Comment:
can we just replace the whole config if there is a mismatch, why do we have to cherry-pick specific stuff? Just replace the config and and restart the brokers instances?